### PR TITLE
For the python package workflow, split building from uploading.

### PR DESCRIPTION
**Pull request by stevenbronson-wk**
#### Steps to Test
Verify that this doesn't prompt for a release deployment.  Maybe test the twine upload.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by austinmatherne-wk — APPROVED** [review]

**Review by derekgengenbacher-wf — DISMISSED** [review]

**Review by aaroncameron-wk — DISMISSED** [review]

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -24,14 +23,27 @@ jobs:
         run: |
           pip install setuptools wheel twine build
```
Comment (austinmatherne-wk) [inline .github/workflows/python-package.yml:24]: This runner doesn't use twine.
Preexisting, I think you need to use the `-U` param to update setuptools and wheel:
```suggestion
          pip install -U setuptools wheel
          pip install build
```

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -24,14 +23,27 @@ jobs:
...
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
       - name: Publish package on release
-        if: startsWith(github.ref, 'refs/tags')
         env:
           TWINE_USERNAME: ${{ secrets.TWINE_USERNAME }}
           TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
-        run: twine upload dist/*
+        run: twine upload */*
```
Comment (austinmatherne-wk) [inline .github/workflows/python-package.yml:49]: The runner for this job doesn't have twine or python (you can test this on a fork using `twine upload --repository testpypi`).

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -24,14 +23,27 @@ jobs:
...
+          path: dist/*.whl
+
+  publish-package:
+    if: startsWith(github.ref, 'refs/tags')
+    needs: build-package
+    environment: release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
```
Comment (austinmatherne-wk) [inline .github/workflows/python-package.yml:44]: Use the full action version and let dependabot manage updates:
```suggestion
        uses: actions/download-artifact@v3.0.0
```

**Review by aaroncameron-wk — APPROVED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Comment by austinmatherne-wk** [comment]
# Verified

* pytohn package
* tagged release job to test pypi

**Pull request merged**